packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+                }
+                if (x > 0 && x < canvasWidth - size && (y < canvasHeight - size || y > 0)) {
+                    y = y > (canvasHeight - size) / 2 ? canvasHeight - size : 0;
+                }
+
+                const clusterNode = this.getClusterRendering(`cluster-${i}-proxy`, numProxiesInCluster, size, x, y);
+                res.push({
+                    node: clusterNode,
+                    transform: {
+                        x: x, y: y, scale: 1, width: size, height: size,
+                        numProxies: numProxiesInCluster // Store the number of proxies in this cluster in case the cluster is clustered later on
+                    } as any as TransformAttributes
+                });
+            }
+
+            // Filter all overlapping nodes
+            res = res.filter((_, index) => !overlapIndexGroups.reduce((acc, group) => acc.concat(group), []).includes(index));
+        }
+
+        return res;
+    }
+
+    /** Returns the proxy rendering for a single off-screen node and applies logic, e.g. the proxy's position. */
+    private createSingleProxy(node: SKNode | VNode, transform: TransformAttributes, ctx: SKGraphModelRenderer): VNode | undefined {
+        let transformString = `translate(${transform.x}, ${transform.y})`;
+        if (transform.scale) {
+            transformString += ` scale(${transform.scale})`;
+        }
+
+        if (!(node instanceof SKNode)) {
+            // VNode, this is a predefined rendering (e.g. cluster)
+            this.updateTransform(node, transformString);
+            return node;
+        }
+
+        // Get VNode
+        const id = this.getProxyId(node.id);
+        let vnode = this.renderings.get(id);
+        if (!vnode) {
+            // Node hasn't been rendered yet (cache empty)
+
+            // Clone the node
+            const clone: SKNode = Object.create(node);
+            // Change its id for good measure
+            clone.id = id;
+            // Clear children, proxies don't show nested nodes (but keep Labels)
+            clone.children = clone.children.filter(node => !(node instanceof SKNode || node instanceof SKEdge));
+            // Update bounds
+            clone.bounds = transform;
+
+            // Check if synthesis has specified a proxy rendering
+            if (this.useSynthesisProxyRendering && node.properties && node.properties[ProxyView.PROXY_RENDERING_PROPERTY]) {
+                // Proxy rendering available
+                console.log("node"); // FIXME:
+                console.log(node.properties[ProxyView.PROXY_RENDERING_PROPERTY]); // FIXME:
+                console.log("proxyRendering"); // FIXME:
+                console.log(node.properties[ProxyView.PROXY_RENDERING_PROPERTY + "2"]); // FIXME:
+                clone.data = node.properties[ProxyView.PROXY_RENDERING_PROPERTY] as KGraphData[];
+            } else {
+                // Fallback, use mock
```

Comment:
careful with overusing the word mock here, in this case it is simply the client-side default/fallback handling, mock would either refer to the mockup (design) or a mock object (standin during development of unimplemented interfaces)